### ISSUE TITLE
[BUG] NAT iptables return inconsistent

### ISSUE THREAD

**Issue by dennybaa** [sshuttle/sshuttle#247]
Sshutle creates a nat table ruble `iptables -t nat -D ${SSHUTTLE_CHAIN} -d 127.0.0.1/32 -p tcp -j RETURN` this is not that actually required. What happens is that all the traffic is piped into tunnel expect traffic targeting localhost.
This _doesn't take into consideration_ that localhost is not only `127.0.0.1` address available locally!

```
Chain sshuttle-12300 (2 references)
target     prot opt source               destination         
RETURN     tcp  --  0.0.0.0/0            127.0.0.1           
```

Consistent configuration would be:

```shell
iptables -t nat -I sshuttle-12300 -m addrtype --dst-type LOCAL -j RETURN
```

PS: Also when running inside a docker container sshutle breaks all the traffic originating from the host to container exposed ports.

**Comment by brianmay** [comment]
The purpose of this rule is to prevent redirecting traffic the port we are listening on. Hence we only need to redirect traffic to 127.0.0.1, because that is the only address we are listening on.

You seem to be indicating however that we should have a more broad rule that prevents intercepting any traffic on the localhost network.

Have I understood correctly?

**Comment by dennybaa** [comment]
Yup right. https://unix.stackexchange.com/questions/130807/what-are-the-definitions-of-addrtype-in-iptables?answertab=active#tab-top

`local - the destinations are assigned to this host.  The packets are looped back and delivered locally.`

The "LOCAL" address match includes any `127.*` ip always, but also takes into account ip addresses directly bound to your network interface.

**Comment by dennybaa** [comment]
@brianmay Hi, if you'd like here's a change for iptables FYI

**Comment by danjeffery** [comment]
I just added [PR 337](https://github.com/sshuttle/sshuttle/pull/337) to fix the tests failing and you should now be able to merge this into the project.
